src/inputs.rs

```diff
@@ -19,25 +19,36 @@ pub fn validate_inputs(config: &Config) -> Result<Config, frost_ed25519::Error>
         return Err(frost_ed25519::Error::InvalidMinSigners);
     }
 
-    Ok(*config)
+    Ok(())
 }
 
-pub fn request_inputs() -> Config {
+pub fn request_inputs(input: &mut impl BufRead) -> Result<Config, frost_ed25519::Error> {
     println!("The minimum number of signers: (2 or more)");
 
     let mut min = String::new();
-    io::stdin().read_line(&mut min).expect("invalid input");
-
-    let min_signers = min.trim().parse::<u16>().expect("Invalid input");
+    input.read_line(&mut min).unwrap();
+
+    let min_signers = min.trim().parse::<u16>();
+    match min_signers {
+        Ok(_) => (),
+        Err(_) => {
+            return Err(frost_ed25519::Error::InvalidMinSigners);
+        }
+    }
```

Comment:
You can make this a bit more idiomatic with something like

```
let min_signers = min.trim().parse::<u16>().map_err(|_| Err(frost_ed25519::Error::InvalidMinSigners))?;
```
(Haven't tested it so it may need some adjustment)

Reply:
Done in [a76acb6](https://github.com/ZcashFoundation/frost-zcash-demo/pull/4/commits/a76acb60c725258f3409ab359ed6b7963988c005)